### PR TITLE
created a feature to sort products by category

### PR DESCRIPTION
**Pull request by Itay-Manevych**

### CODE REVIEW

**Comment by Itay-Manevych** [comment]
@Tamir198 @Br3adMan1 if you have time take a look at my code

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+import DelProduct from './pages/admin/DelProduct';
+import DelCategory from './pages/admin/DelCategory';
+import EditOrder from './pages/admin/EditOrder';
+import EditCategory from './pages/admin/EditCategory';
+import EditProduct from './pages/admin/EditProduct';
+import DelOrder from './pages/admin/DelOrder';
+import Orders from './pages/admin/Orders';
 
 function App() {
+  const [AdminAuth] = useState('123456');
```
Comment (Tamir198) [inline frontend/src/App.jsx:24]: What is the meaning if the initial state value?

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,25 +1,54 @@
...
+            <Route path="/admin/category/:id" element={<Category />} />
+            <Route path="/admin/product/:id" element={<Product />} />
+            <Route path="/admin/editOrder/:id" element={<EditOrder />} />
+            <Route path="/admin/editProduct/:id" element={<EditProduct />} />
+            <Route path="/admin/editCategory/:id" element={<EditCategory />} />
+            <Route path="/admin/delProduct/:id" element={<DelProduct />} />
+            <Route path="/admin/delCategory/:id" element={<DelCategory />} />
+            <Route path="/admin/delOrder/:id" element={<DelOrder />} />
+          </Routes>
+        </BrowserRouter>
```
Comment (Tamir198) [inline frontend/src/App.jsx:49]: Consider using nested rout so you wont have to write /admin on every rout

File: frontend/src/Components/Admin/AdminLogin.jsx [hunk 3]
```diff
@@ -2,9 +2,12 @@ import React from 'react';
...
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
   return (
     <div>
-      <form method="POST">
+      <form method="POST" onSubmit={handleSubmit}>
         <Input type="text" name="username" />
         <Input type="password" name="password" />
         <button type="submit">Login</button>
```
Comment (Tamir198) [inline frontend/src/Components/Admin/AdminLogin.jsx:13]: If there are values that repeating themself you may want to consult with your team and add a constants file for those values.

File: frontend/src/Components/MainPage/CategoryDisplay/CategoryDisplay.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      </select>
+      <ul>
+        {filteredProducts.map((product) => (
+          <div key={product.id}>
+            <h2>{product.name}</h2>
+            <p>{product.description}</p>
+            <p>{product.price}</p>
+            <img src={product.image} alt={product.name} />
+          </div>
+        ))}
```
Comment (Tamir198) [inline frontend/src/Components/MainPage/CategoryDisplay/CategoryDisplay.jsx:28]: This looks like a classic thing to extract into a generic list inside the UI toolkit.

File: frontend/src/Components/MainPage/ProductDisplay.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return (
+    <div>
+      {products.map((product) => (
+        <div key={product.id}>
+          <h2>{product.name}</h2>
+          <p>{product.description}</p>
+          <p>{product.price}</p>
+          <img src={product.image} alt={product.name} />
+        </div>
+      ))}
```
Comment (Tamir198) [inline frontend/src/Components/MainPage/ProductDisplay.jsx:14]: The same comment about the generic list, you could have avoided writing the same code again

File: frontend/src/pages/MainPage/MainPage.jsx [hunk 6]
```diff
@@ -1,8 +1,13 @@
 import React from 'react';
+// import ProductDisplay from '../../Components/MainPage/ProductDisplay';
+import CategoryDisplay from '../../Components/MainPage/CategoryDisplay/CategoryDisplay';
```
Comment (Tamir198) [inline frontend/src/pages/MainPage/MainPage.jsx:3]: I don't think this comment contributes, if you don't need some code just delete it. You can always use git to restore the changes

File: frontend/src/pages/admin/Product.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+function Product() {
+  const AdminAuth = useContext(AdminAuthContext);
+  const navigate = useNavigate();
+  const params = useParams();
+  return (
+    <div>
+      {AdminAuth ? params.id : navigate('/')}
+    </div>
+  );
```
Comment (Tamir198) [inline frontend/src/pages/admin/Product.jsx:13]: This code: 

```
 const AdminAuth = useContext(AdminAuthContext);
 <div>
      {AdminAuth ? params.id : navigate('/')}
    </div>
```

Is repeats itself a lot of times. do you think you can extract it into something more generic?

**Review by Tamir198 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**